website/client/js/controllers/filtersCtrl.js

```diff
@@ -17,6 +17,7 @@ habitrpg.controller("FiltersCtrl", ['$scope', '$rootScope', 'User', 'Shared',
             User.updateTag({params:{id:tag.id}, body:{name:tag.name}});
         })
         $scope._editing = false;
+        User.sync();
```

Comment:
Syncing is a pretty weighty operation. You should be able to pull the necessary tag info from the response of both the update and the create calls without the need to sync.